src/main/java/net/logstash/logback/stacktrace/ShortenedThrowableConverter.java

```diff
@@ -230,16 +248,22 @@ public String convert(ILoggingEvent event) {
         if (throwableProxy == null || isExcludedByEvaluator(event)) {
             return CoreConstants.EMPTY_STRING;
         }
-        
+
+        // compute stack trace hashes
+        Deque<String> stackHashes = null;
```

Comment:
Rather than calculating all the hashes up front, and creating an intermediary `Deque` object, I think it would be simpler (and more efficient) to calculate hashes of individual throwables as the converter is walking through the cause chain.

i.e Don't calculate the hashes here.  Don't create a Deque object.  Just call the StashHasher in `appendFirstLine`.  In this case, StashHasher would have a method that calculates the hash of the throwable passed in as an argument without taking its causes into consideration.